packages/interface-adapter/lib/adapter/web3/index.ts

```diff
@@ -80,9 +82,13 @@ export class Web3InterfaceAdapter implements InterfaceAdapter {
     const gas = new BN(receipt.gasUsed);
     const value = new BN(tx.value);
     const cost = gasPrice.mul(gas).add(value);
+    const timestamp =
+      typeof block.timestamp === "string"
+        ? parseInt(block.timestamp)
+        : block.timestamp;
```

Comment:
No idea, sorry. I just handled the case because the type check failed as it was a `string | BN` being assigned to a `BN`.

Reply:
On further reflection, I think it's because the return type for `web3.eth.getBlock` is also used elsewhere in web3 as an input type, and they accept both `string` and `BN` inputs. Chances are that it's always returned as a BN, however.